chia/wallet/wallet_state_manager.py

```diff
@@ -416,86 +416,101 @@ async def create_more_puzzle_hashes(
-
-            # If the key was replaced (from_zero=True), we should generate the puzzle hashes for the new key
-            if from_zero:
-                start_index = 0
-            last_index = unused + to_generate
-            if start_index >= last_index:
-                self.log.debug(f"Nothing to create for for wallet_id: {wallet_id}, index: {start_index}")
+                if last + 1 >= last_index:
+                    self.log.debug(f"Nothing to create for for wallet_id: {wallet_id}, index: {start_index}")
+                    continue
+                start_index = min(start_index, last + 1)
+                start_index_by_wallet[wallet_id] = last + 1
             else:
-                creating_msg = (
-                    f"Creating puzzle hashes from {start_index} to {last_index - 1} for wallet_id: {wallet_id}"
-                )
-                self.log.info(f"Start: {creating_msg}")
-                if self.private_key is not None:
-                    intermediate_sk = master_sk_to_wallet_sk_intermediate(self.private_key)
-                intermediate_pk_un = master_pk_to_wallet_pk_unhardened_intermediate(self.root_pubkey)
-                for index in range(start_index, last_index):
-                    if target_wallet.type() == WalletType.POOLING_WALLET:
-                        continue
+                start_index_by_wallet[wallet_id] = 0
 
-                    if self.private_key is not None:
-                        # Hardened
-                        pubkey: G1Element = _derive_path(intermediate_sk, [index]).get_g1()
-                        puzzlehash: bytes32 = target_wallet.puzzle_hash_for_pk(pubkey)
-                        self.log.debug(f"Puzzle at index {index} wallet ID {wallet_id} puzzle hash {puzzlehash.hex()}")
-                        new_paths = True
-                        derivation_paths.append(
-                            DerivationRecord(
-                                uint32(index),
-                                puzzlehash,
-                                pubkey,
-                                target_wallet.type(),
-                                uint32(target_wallet.id()),
-                                True,
-                            )
-                        )
-                    # Unhardened
-                    pubkey_unhardened: G1Element = _derive_pk_unhardened(intermediate_pk_un, [index])
-                    puzzlehash_unhardened: bytes32 = target_wallet.puzzle_hash_for_pk(pubkey_unhardened)
-                    self.log.debug(
-                        f"Puzzle at index {index} wallet ID {wallet_id} puzzle hash {puzzlehash_unhardened.hex()}"
-                    )
-                    # We await sleep here to allow an asyncio context switch (since the other parts of this loop do
-                    # not have await and therefore block). This can prevent networking layer from responding to ping.
-                    await asyncio.sleep(0)
```

Comment:
Do we not need this anymore?  Is it because you expect this function to be much faster or some other reason?

Reply:
yeah, maybe I don't have the strongest justification for removing this. In my profiling of this, the bulk of the time is spent deriving the keys, and the keys are no longer derived in this loop. Computing the puzzle hashes are also somewhat expensive, but I'm optimizing this in other PRs, so it's an increasingly small part of the problem.

Also, sleeping *every* turn through the loop seems quite excessive. It probably slows down the total wall-clock time completing this. If the problem of other tasks being held up comes back, I think we should solve it in a better way. At the very least only yield every 100 keys, or something like that.